### PR TITLE
BUGZ-355: Persistent UI Element

### PR DESCRIPTION
**Pull request by PrestonB1123**
fixed persistent UI element when switching from HMD to Desktop. Not a complete fix but sufficient for M1.
https://highfidelity.atlassian.net/browse/BUGZ-355

### CODE REVIEW

File: diff.patch [hunk 1]
```diff
@@ -0,0 +1,40 @@
+diff --git a/scripts/simplifiedUI/ui/simplifiedUI.js b/scripts/simplifiedUI/ui/simplifiedUI.js
```
Comment (zfox23) [inline diff.patch]: please remove this file!

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -978,7 +978,7 @@ Setting::Handle<int> sessionRunTime{ "sessionRunTime", 0 };
 
 const float DEFAULT_HMD_TABLET_SCALE_PERCENT = 60.0f;
 const float DEFAULT_DESKTOP_TABLET_SCALE_PERCENT = 75.0f;
-const bool DEFAULT_DESKTOP_TABLET_BECOMES_TOOLBAR = true;
+const bool DEFAULT_DESKTOP_TABLET_BECOMES_TOOLBAR = false;
```
Comment (zfox23) [inline interface/src/Application.cpp]: please revert this change

File: interface/src/Application.cpp [hunk 3]
```diff
@@ -9301,6 +9301,7 @@ void Application::updateSystemTabletMode() {
         if (isHMDMode()) {
             DependencyManager::get<TabletScriptingInterface>()->setToolbarMode(getHmdTabletBecomesToolbarSetting());
         } else {
+            //_desktopTabletBecomesToolbarSetting.set(false);
```
Comment (zfox23) [inline interface/src/Application.cpp]: please revert this change

File: scripts/simplifiedUI/ui/simplifiedUI.js [hunk 4]
```diff
@@ -457,9 +457,15 @@ function onGeometryChanged(rect) {
...
-function ensureFirstPersonCameraInHMD(isHMDMode) {
+function onDisplayModeChanged(isHMDMode) {
     if (isHMDMode) {
         Camera.setModeString("first person");
+    } else {
+        //works for now, but not a permanent fix by any means.
+        Script.setTimeout(function () {
+            var toolbar = Toolbars.getToolbar(TOOLBAR_NAME);
+            toolbar.writeProperty("visible", false);
+        }, 700);
```
Comment (zfox23) [inline scripts/simplifiedUI/ui/simplifiedUI.js]: please redefine this magic number as something like `var TIMEOUT_BEFORE_REHIDE_TOOLBAR_MS = 700` above the definition of this function.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19927/HighFidelity-Beta-PR15745-b32a715.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19927/HighFidelity-Beta-Interface-PR15745-b32a715.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19927/HighFidelity-Beta-PR15745-b32a715.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19927/HighFidelity-Beta-Interface-PR15745-b32a715.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19927/HighFidelity-Beta-PR15745-b32a715.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19927/HighFidelity-Quest-Beta-PR15745-b32a715.apk)

File: interface/src/Application.cpp (hunk 2 above)
Comment (PrestonB1123) [reply interface/src/Application.cpp]: Ack

File: diff.patch (hunk 1 above)
Comment (PrestonB1123) [reply diff.patch]: Ack

File: interface/src/Application.cpp (hunk 3 above)
Comment (PrestonB1123) [reply interface/src/Application.cpp]: Ack

File: scripts/simplifiedUI/ui/simplifiedUI.js (hunk 4 above)
Comment (PrestonB1123) [reply scripts/simplifiedUI/ui/simplifiedUI.js]: Ack

File: scripts/simplifiedUI/ui/simplifiedUI.js [hunk 5]
```diff
@@ -457,9 +457,16 @@ function onGeometryChanged(rect) {
     }
 }
 
-function ensureFirstPersonCameraInHMD(isHMDMode) {
+var TIMEOUT_BEFORE_REHIDE_TOOLBAR_MS = 700
+function onDisplayModeChanged(isHMDMode) {
     if (isHMDMode) {
         Camera.setModeString("first person");
+    } else {
```
Comment (zfox23) [inline scripts/simplifiedUI/ui/simplifiedUI.js]:
```suggestion
    } else if (Settings.getValue("simplifiedUI/keepExistingUIAndScripts", false)) {
```

Sorry, I forgot about this yesterday!

File: scripts/simplifiedUI/ui/simplifiedUI.js [hunk 6]
```diff
@@ -457,9 +457,16 @@ function onGeometryChanged(rect) {
...
+        //works for now, but not a permanent fix by any means.
```
Comment (zfox23) [inline scripts/simplifiedUI/ui/simplifiedUI.js]:
```suggestion
        // works for now, but not a permanent fix by any means.
```

And while you're at it, please add this space here for style consistency!

File: scripts/simplifiedUI/ui/simplifiedUI.js (hunk 5 above)
Comment (PrestonB1123) [reply scripts/simplifiedUI/ui/simplifiedUI.js]: Ack

File: scripts/simplifiedUI/ui/simplifiedUI.js (hunk 6 above)
Comment (PrestonB1123) [reply scripts/simplifiedUI/ui/simplifiedUI.js]: Ack

File: scripts/simplifiedUI/ui/simplifiedUI.js (hunk 5 above)
Comment (PrestonB1123) [reply scripts/simplifiedUI/ui/simplifiedUI.js]: Ack

File: scripts/simplifiedUI/ui/simplifiedUI.js (hunk 6 above)
Comment (PrestonB1123) [reply scripts/simplifiedUI/ui/simplifiedUI.js]: Ack

File: scripts/simplifiedUI/ui/simplifiedUI.js [hunk 7]
```diff
@@ -457,9 +457,16 @@ function onGeometryChanged(rect) {
     }
 }
 
-function ensureFirstPersonCameraInHMD(isHMDMode) {
+var TIMEOUT_BEFORE_REHIDE_TOOLBAR_MS = 700
```
Comment (zfox23) [inline scripts/simplifiedUI/ui/simplifiedUI.js]:
```suggestion
var TIMEOUT_BEFORE_REHIDE_TOOLBAR_MS = 700;
```

Comment (PrestonB1123) [reply scripts/simplifiedUI/ui/simplifiedUI.js]: Ack

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19937/HighFidelity-Beta-PR15745-6c0bbfa.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/19937/HighFidelity-Quest-Beta-PR15745-6c0bbfa.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19937/HighFidelity-Beta-PR15745-6c0bbfa.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/19937/HighFidelity-Beta-Interface-PR15745-6c0bbfa.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19937/HighFidelity-Beta-PR15745-6c0bbfa.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/19937/HighFidelity-Beta-Interface-PR15745-6c0bbfa.exe
